etc/systemd/system-generators/zfs-mount-generator.in

```diff
@@ -37,7 +37,6 @@ do_fail() {
 # test if $1 is in space-separated list $2
 is_known() {
   query="$1"
-  IFS=' '
```

Comment:
Who guarantees that the script will be run with a sane IFS (by our definition)? (I know, 99% of runs will be started by systemd, which will not set IFS, but people being people, someone's foot is bound to get hurt.)

What's the gain here vs saving and restoring? We got a lot of safety "for free" with the explicit approach.

For example, running the generator with $IFS not containing a space and/or containing a hyphen would surely confuse the generator in many amusing ways.

If you really, really want to go forth this change anyway, I'd suggest at least adjusting the function description accordingly.

Reply:
I really don't understand the concern. IFS is set by `/bin/sh`, it cannot be set by just `export`ing `IFS`

Reply:
Ok, I disagree for a different reason: if you set IFS, you must have wanted it as such, and we should respect that.  systemd presumably doesn't set it (because that would be crazy), but if there's some reason to be worried about this, we should set it at the top.

Reply:
Why should we respect it? The goal of is_known is not to check $2 as a IFS-separated list but as a space-separated one. The separation character is purposely only one, not a list, so no accidental splits happen, given no spaces are used in items.

Saving and restoring the IFS after we're done is the way to respect the IFS.

Reply:
I look at this as being: we set IFS to tab only to tab-split the list that can contain spaces in items. Other than that, let’s behave as a normal shell script. That way we aren’t surprising. Note that my change didn’t work because the existing code was not consistent with normal shell scripts. If/when this is ported to C, then we will have to be explicit about splitting.

Reply:
I don't get what's non-standard about stashing away IFS, setting it to the locally desired space-only value, doing our thing, and then setting it back to the stashed value, all neatly contained in a function.
Splitting according to global IFS is not desirable in this function, why do it?

If you wanted to be pedantic, you could redefine `is_known` to work on an IFS-split list, but then I'd insist on calling it like `IFS=' ' is_known "$unit" "$list"`, given how the list is constructed.

I don't see the value in any of this.

Reply:
To be clear:
> Note that my change didn’t work because the existing code was not consistent with normal shell scripts.

I'm not sure what exactly this part means. Overriding without saving and restoring IFS in is_known was a bug, yes, but removing everything IFS-related from the function doesn't seem like the appropriate solution.

> If/when this is ported to C, then we will have to be explicit about splitting.

I chose the explicit space-separation wording and code to make this part somewhat robust by (enforcing the) contract, not purely out of necessity.

Reply:
I was going to argue against this, but then I realized that I'm still not doing this correctly. We need to set IFS to a newline for the pool splitting, as pools can have spaces in their names.

BTW, there apparently is no portable way to set IFS to a newline other than a literal newline in the source.

Reply:
I've pushed a version that sets IFS explicitly in every case. Since I'm setting it explicitly every time, I removed any resetting of it. I also moved the `set -f` outside of the functions, since it is not actually contained to the function and we are never doing `set +f`. I had to have the `set -f` inside the main for loop, as we do need globbing for the cache files.